x-pack/plugins/security_solution/public/common/store/sourcerer/reducer.ts

```diff
@@ -45,6 +46,12 @@ export const sourcererReducer = reducerWithInitialState(initialSourcererState)
       ...dataView,
     })),
   }))
+  .case(updateSourcererDataViews, (state, { dataView }) => ({
```

Comment:
do you also need to check if the data view is the default data view and update that if it is? I think the current `setDataView` action achieves what you want here

Reply:
you tell me because the type of this two different actions are different and I can combine them and everything will be optional. I do not think that we will need to update the default data view because they are different attributes.